src/app/application.cpp

```diff
@@ -242,6 +242,48 @@ void Application::processMessage(const QString &message)
         m_paramsQueue.append(params);
 }
 
+void Application::runExternalProgram(BitTorrent::TorrentHandle *const torrent) const
+{
+    QString program = Preferences::instance()->getAutoRunProgram();
+    program.replace("%N", torrent->name());
+    program.replace("%L", torrent->category());
+    program.replace("%F", Utils::Fs::toNativePath(torrent->contentPath()));
+    program.replace("%R", Utils::Fs::toNativePath(torrent->rootPath()));
+    program.replace("%D", Utils::Fs::toNativePath(torrent->savePath()));
+    program.replace("%C", QString::number(torrent->filesCount()));
+    program.replace("%Z", QString::number(torrent->totalSize()));
+    program.replace("%T", torrent->currentTracker());
+    program.replace("%I", torrent->hash());
+
+    Logger *logger = Logger::instance();
+    logger->addMessage(tr("Torrent: %1, running external program, command: %2").arg(torrent->name()).arg(program));
+
+#if defined(Q_OS_UNIX)
+    QProcess::startDetached(QLatin1String("/bin/sh"), {QLatin1String("-c"), program});
```

Comment:
@sledgehammer999 
>Thus, if this code fails, system() fails too in almost every Linux distribution.

I think we can rely on this. Or... revert to `QProcess::startDetached(program);` and let user prepend the path to shell (in the textbox) himself.

>However, POSIX states that this is not guaranteed. Perhaps, we can use 'sh' here and rely on PATH?

no support from `QProcess::startDetached` and I don't want to search for `sh` in `PATH` myself.